File: get_volume.py
import subprocess
import re
from time import sleep

def get_volume_level():
    try:        
        result = subprocess.run(['pactl', 'get-sink-volume', '@DEFAULT_SINK@'],
                                capture_output=True, text=True, check=True)
        output = result.stdout.strip()
        match = re.search(r'(\d+)%', output)
        if match:
            return int(match.group(1))
        else:
            return 0
    except subprocess.CalledProcessError:
        return 0

while True:
    print(f"{get_volume_level()}%", flush=True)
    sleep(0.2)

File: get_volume.lua
local function get_volume_level()
    -- Run the pactl command to get the volume level
    local handle = io.popen("pactl get-sink-volume @DEFAULT_SINK@")
    local result = handle:read("*all")
    handle:close()

    -- Extract the volume percentage using string patterns
    local volume = result:match("(%d+)%%")
    if volume then
        return tonumber(volume)
    else
        return 0
    end
end

-- Loop to continuously print the volume level every 0.2 seconds
while true do
    print(get_volume_level() .. "%")
    os.execute("sleep 0.1")
end
